js/views/modals/EditListing.js

```diff
@@ -0,0 +1,478 @@
+      }
+
+      if (!_.isEqual(this.model.toJSON(), this._origModel.toJSON())) {
+        this._origModel.set(this.model.toJSON(), { silent: true });
+
+        // A change events won't fire on a parent model if a nested model change.
+        // The nested models would need to have change events manually bound to them
+        // which is cumbersome with a model like this with so many levels of nesting.
+        // So, for now, we'll manually fire a change event if anything has changed.
+        // TODO: Find a reasonable way to manage something like this and
+        // put it in the baseModel.
+        this._origModel.trigger('change', this._origModel);
+      }
+    });
+
+    this.innerListing = this.model.get('listing');
+    this.selectedNavTabIndex = 0;
+    this.photoUploads = new Collection();
+    this.createMode = !(this.model.lastSyncedAttrs.listing &&
+      this.model.lastSyncedAttrs.listing.slug);
+    this.photoUploads = [];
+    this.images = this.innerListing.get('item').get('images');
+
+    loadTemplate('modals/editListing/uploadPhoto.html',
+      uploadT => (this.uploadPhotoT = uploadT));
+
+    this.listenTo(this.images, 'add', this.onAddImage);
+    this.listenTo(this.images, 'remove', this.onRemoveImage);
+  }
+
+  className() {
+    return `${super.className()} editListing tabbedModal modalTop`;
+  }
+
+  events() {
+    return {
+      'click .js-scrollLink': 'onScrollLinkClick',
+      'click .js-save': 'onSaveClick',
+      'change #editListingType': 'onChangeListingType',
+      'change #editListingSlug': 'onChangeSlug',
+      'change .js-price': 'onChangePrice',
+      'change #inputPhotoUpload': 'onChangePhotoUploadInput',
+      'click .js-addPhoto': 'onClickAddPhoto',
+      'click .js-removeImage': 'onClickRemoveImage',
+      'click .js-cancelPhotoUploads': 'onClickCancelPhotoUploads',
+      ...super.events(),
+    };
+  }
+
+  get MAX_PHOTOS() {
+    return this.model.get('listing').get('item').maxImages;
+  }
+
+  onAddImage(image) {
+    const imageHtml = this.uploadPhotoT({
+      closeIconClass: 'js-removeImage',
+      ...image.toJSON(),
+    });
+
+    this.$photoUploadItems.append(imageHtml);
+  }
+
+  onRemoveImage(image, images, options) {
+    // 1 is added to the index to account for the .addElement
+    this.$photoUploadItems.find('li')
+      .eq(options.index + 1)
+      .remove();
+  }
+
+  onClickRemoveImage(e) {
+    // since the first li is the .addElement, we need to subtract 1 from the index
+    const removeIndex = $(e.target).parents('li').index() - 1;
+
+    this.images.remove(this.images.at(removeIndex));
+  }
+
+  onClickCancelPhotoUploads() {
+    this.inProgressPhotoUploads.forEach(photoUpload => photoUpload.abort());
+  }
+
+  onChangePrice(e) {
+    let updatedVal = $(e.target).val().trim();
+    const valAsNumber = Number(updatedVal);
+
+    if (!isNaN(valAsNumber)) {
+      updatedVal = valAsNumber.toFixed(2);
```

Comment:
If your currency is BTC, you need to be able to enter prices as small as 0.0005, this is truncating the price to zero if it's lower than 0.01.